docs/references/kontena-yml.md

```diff
@@ -159,6 +159,24 @@ affinity:
 
 #### hooks
 
+**pre_start**
+
+`pre_start` hooks define commands that are executed before a container is started on a node. If multiple hooks are provided they are executed in the order defined. Note that these commands are executed *before* `wait_for_port` check.
```

Comment:
* If any pre_start hook fails, then the remaining hooks will be skipped, and the container will not be created
* Any changes to the container filesystem from pre-start hooks will be reset when starting the actual container
* The pre-start hooks are run with a different overlay network IP address, and cannot be used to e.g. register the container IP somewhere (use post-start hooks)